Add monthly featured developer showcase

Aashish-Jha-11 opened the pull request:
Highlights top GitHub developer each month based on engagement (followers, stars, repos, sponsors, activity). Features beautiful gradient banner with user stats and top languages. Auto-updates when running fetch.py.

**Current featured:** Linus Torvalds (engagement score: 74.97)

**What's included:**
- 🎯 Smart engagement scoring algorithm (40% followers, 30% stars, 15% repos, 10% sponsors, 5% recent activity)
- 🎨 Beautiful gradient banner with glassmorphism effects
- 📱 Fully responsive design (mobile, tablet, desktop)
- 🔄 Automatically updates when running `fetch.py`
- ⚡ Quick manual generation via `generate_featured.py`

Resolves #122

BaseMax commented:
Descriptions are not readable well:
<img width="810" height="21" alt="image" src="https://github.com/user-attachments/assets/81234e73-0d70-431f-a8d2-4665529dbdc1" />

Aashish-Jha-11 commented:
@BaseMax Sir! I think Keeping JSONs separate is better:

Both load in parallel Promise.all & featured.json
No scroll jumping - featured section loads immediately
Better cache management - can update featured monthly without invalidating main cache

BaseMax commented:
> Better cache management - can update featured monthly without invalidating main cache

Did you handled it in manifest.json or service-worker.js file?

BaseMax commented on docs/styles.css:
```diff
@@ -188,10 +188,13 @@ h1 {
-    color: rgba(255, 255, 255, 0.95);
-    font-weight: 600;
+    font-size: 1.1rem;
+    color: #ffffff !important;
```
Do you nou need to remove `!important`?

BaseMax commented:
> Should we add explicit cache handling in service-worker.js? We could set different cache durations (1 month for featured.json vs longer for users.json).

Yes.

BaseMax commented:
This branch has not been deployed
No deployments

BaseMax commented:
Fully ready?

jbampton commented:
This branch has conflicts that must be resolved

BaseMax commented:
This branch has conflicts that must be resolved

BaseMax commented:
We need to close and merge it faster. @Aashish-Jha-11 please do action.

Aashish-Jha-11 commented:
Hi @BaseMax,

Resolved merge conflicts and fixed all critical issues in our feature code. Latest commit: 4e08e3b

Fixed:
✅ Python: All logging f-strings converted to parameter substitution, jinja2 autoescape enabled
✅ JavaScript: Strict equality (=== instead of ==) in our feature code

Still failing (DeepSource):
The remaining issues are in Bootstrap vendor files (jquery.min.js, bootstrap.js, popper.min.js) that were reformatted by prettier:

JS-0125 (Undeclared variables) - jQuery/Bootstrap patterns
JS-0128 (Unused variables) - Vendor code
JS-0103 (IIFE wrapping) - jQuery patterns
Others - All in vendor libraries
Our feature code (script.js, service-worker.js, fetch.py, render.py) is clean and follows best practices.

Options:

Exclude vendor files from DeepSource analysis
Revert vendor file formatting (keep them minified)
Merge as-is since issues are in external dependencies
Ready for your decision.

BaseMax commented:
[docs/bootstrap/css/bootstrap-grid.css‎](https://github.com/john-bampton/john-bampton.github.io/pull/124/changes#diff-64aed873b45cb0a6f9dcbeb8b6e142728eefdf79355c78208a9ce169c13a1337)

all bootstrap assets files should be minify.

BaseMax commented:
If we have both .min.css and nonmin.css so please remove nonmin and just keep using .min.css or .min.js.

BaseMax commented:
you should keep [‎docs/featured.json‎](https://github.com/john-bampton/john-bampton.github.io/pull/124/changes#diff-f5addc2562796ab9006be402d2c85e64829bdd314ba9b0199af422e84c3c1d9b) file minified.

BaseMax commented:
you should keep [docs/manifest.json‎](https://github.com/john-bampton/john-bampton.github.io/pull/124/changes#diff-0cbcfff19a4479219e722863d8f59891658d042d3fac32fe9d9a75d650ce0524) file minified. or maybe this comes from pre-commit or something?

BaseMax commented:
You removed some code from html which are good to keep.

<img width="1198" height="587" alt="image" src="https://github.com/user-attachments/assets/a2a5f9e4-3d4b-4b84-9b16-8104862f7ea9" />


such as
```
                <option value="followers-growth-desc">
                  Trending (Followers ↑)
                </option>
                <option value="followers-growth-asc">
                  Trending (Followers ↓)
                </option>
```

and 

```
            <button class="btn-apply" onclick="exportFilteredJSON()">
              Export JSON
            </button>
            <button class="btn-apply" onclick="exportFilteredCSV()">
              Export CSV
            </button>
```

BaseMax commented:
You have removed most codes in fetch.py and render.py such as:

<img width="723" height="735" alt="image" src="https://github.com/user-attachments/assets/50cb7501-b046-4eb3-92fc-cd74c99145d6" />


Next:

variable name should be log, not logger. you changed my code again too:
<img width="1424" height="293" alt="image" src="https://github.com/user-attachments/assets/9df0a1ef-c4fa-48fc-ae80-791e56d736f1" />


You removed other codes:
<img width="708" height="707" alt="image" src="https://github.com/user-attachments/assets/84883b27-f6f7-47d8-b084-da82865261b3" />

BaseMax commented:
If it's hard for you to check/fix conflicts with pay attention to details. i suggest you to close this pr and open a new one and just copy paste your important/new changes

Aashish-Jha-11 commented:
Closing this PR due to merge conflict damage. Replaced with clean implementation in #187
